### PR TITLE
Fix TimeAggregate member types

### LINKED ISSUE #76: Consider updating types of TimeAggregate.{integral,active_time}
This has been sitting in the code as a TODO:

https://github.com/metricq/metricq-python/blob/8e73ceb203a70f5ae8093c218e1eb11f934ea93a/metricq/types.py#L545-L550

Tracking here for visibility.

### PR DESCRIPTION
**Pull request by phijor**
This implements/resolves #76.

Todo:

* [x] Make `TimeAggregate.active_time` a `Timedelta`
* [x] Figure out what to do with `TimeAggregate.integral`:
    - deprecate `integral`, document accordingly
    - add `integral_s`/`integral_ns`
    - document all new properties

### CODE REVIEW

File: metricq/types.py [hunk 1]
```diff
@@ -264,10 +288,10 @@ def __sub__(self, other: Union["Timedelta", "Timestamp", datetime.timedelta]):
             "invalid type to subtract from Timedelta: {}".format(type(other))
         )
 
-    def __truediv__(self, factor):
+    def __truediv__(self, factor) -> "Timedelta":
```
Comment (bmario) [inline metricq/types.py:291]: No typing for factor?

File: metricq/types.py [hunk 2]
```diff
@@ -264,10 +288,10 @@ def __sub__(self, other: Union["Timedelta", "Timestamp", datetime.timedelta]):
...
         return Timedelta(self._value // factor)
 
-    def __mul__(self, factor):
+    def __mul__(self, factor) -> "Timedelta":
```
Comment (bmario) [inline metricq/types.py:294]: No typing for factor?

**Review by bmario — APPROVED** [review]

**Review by bmario — CHANGES_REQUESTED** [review]
Dang it, I didn't meant approve 😠

File: metricq/types.py (hunk 1 above)
Comment (phijor) [reply metricq/types.py:291]: Well... this should be `float`, right?

Comment (tilsche) [reply metricq/types.py:291]: - The internal type should always be an `int`
- We should allow `float` and `int` as parameter (type `float`)
- The previous point should be tested.

Comment (phijor) [reply metricq/types.py:291]: I guess this will be its own PR. I found a few more things that might need refactoring and are related to `Timedelta`s.

**Review by tilsche — APPROVED** [review]

**Review by bmario — APPROVED** [review]

**Pull request merged**